include/ur_api.h

```diff
@@ -1961,6 +1963,36 @@ urTearDown(
     void* pParams                                   ///< [in] pointer to tear down parameters
     );
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Retrieve string representation of the underlying adapter specific
+///        result reported by the the last API that returned
+///        UR_RESULT_ADAPTER_SPECIFIC. Allows for an adapter independent way to
+///        return an adapter specific result.
+/// 
+/// @details
+///     - The string returned via the pMessage is a NULL terminated C style
+///       string.
+///     - The string returned via the pMessage is thread local.
+///     - The entry point will return UR_RESULT_SUCCESS if the result being
+///       reported is to be considered a warning. Any other result code returned
+///       indicates that the adapter specific result is an error.
+///     - The memory in the string returned via the pMessage is owned by the
+///       adapter.
+///     - The application may call this function from simultaneous threads.
+///     - The implementation of this function should be lock-free.
+/// 
+/// @returns
+///     - ::UR_RESULT_SUCCESS
+///     - ::UR_RESULT_ERROR_UNINITIALIZED
+///     - ::UR_RESULT_ERROR_DEVICE_LOST
+///     - ::UR_RESULT_ERROR_INVALID_NULL_POINTER
+///         + `NULL == pMessage`
+UR_APIEXPORT ur_result_t UR_APICALL
+urGetLastResult(
+    const char* pMessage                            ///< [out] optional pointer to a string containing adapter specific result
```

Comment:
In PI this is `char**`, I don't think this will work. Also why is it optional?

Reply:
Added extra `*`. Removed optional.